### PR TITLE
Add optional package-name argument to run-specs grunt task

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
This way we can do `grunt run-specs:markdown-preview` to debug a single package's specs on CI.

### CODE REVIEW

**Comment by kevinsawicki** [comment]
Just curious, does `apm test` not work on CI if you cd into `node_modules/markdown-preview`?

**Comment by maxbrunsfeld** [comment]
I didn't try because I couldn't find `apm`. Looking for it on the CI box now.

**Comment by maxbrunsfeld** [comment]
It just outputs `Tests Failed` for me. Is there a flag I can pass?

**Comment by kevinsawicki** [comment]
It is in `apm/node_modules/.bin/apm` inside the atom repo working directory.

File: build/tasks/spec-task.coffee [hunk 1]
```diff
@@ -106,27 +112,37 @@ module.exports = (grunt) ->
...
-      method = async.series
+    if packageName?
+      runSinglePackageSpecs packageName, (error) ->
+        elapsedTime = Math.round((Date.now() - startTime) / 100) / 10
+        grunt.log.ok("Total spec time: #{elapsedTime}s using #{concurrency} cores")
+        grunt.log.error("[Error]".red + " #{packageName} spec(s) failed") if error?
+        if process.platform is 'win32' and process.env.JANKY_SHA1
+          done()
+        else
+          done(!error?)
```
Comment (kevinsawicki) [inline build/tasks/spec-task.coffee]: Hmm, I thought `done(error)` would work here, does it need to be coerced into a boolean?

**Comment by maxbrunsfeld** [comment]
I'm guessing I need to pass the path to the `atom` command for it to work.

**Comment by kevinsawicki** [comment]
Yeah, probably need to pass in the atom path

**Comment by maxbrunsfeld** [comment]
So it would seem like the Atom path would be in [a fixed location in `/tmp`](https://github.com/atom/atom/blob/master/build/Gruntfile.coffee#L40), but I'm finding that it's in a random temp directory like `/private/var/folders/rj/620tf4yn6pn9q8hj453s2j3r0000gp`.

Is there a better way to do this than this?

```
../../apm/node_modules/.bin/apm test --path $(find /private/var/folders/ -name Atom 2>/dev/null -print -quit)
```

**Comment by kevinsawicki** [comment]
> Is there a better way to do this than this?

Probably not, you could always build to a consistent directory using `--build-dir` but that is pretty unpleasant too.

**Comment by maxbrunsfeld** [comment]
Ok if the `Atom` binary isn't in a stable location on CI, then this probably won't be that helpful in terms of saving time debugging, since you'd have to run `build` in the same grunt session to be able to find it.

**Pull request closed without merging**